src/StoreClient.h

```diff
@@ -67,17 +68,23 @@ class store_client
 public:
     store_client(StoreEntry *);
     ~store_client();
-    bool memReaderHasLowerOffset(int64_t) const;
+    /// \returns the smallest of the passed value and the memory offset of the client
+    int memHeaderOffsetLowerThan(const int64_t offset) const;
     int getType() const;
     void fail();
     void callback(ssize_t len, bool error = false);
+    void callbackClientSide();
```

Comment:
Please document this new method, especially since it is public.

Reply:
This method invokes the previously stored (by store_client::copy()) store_client::Callback::callback_handler. I could document this method a sort of this, but the problem is that none of these referred fileds/methods are documented.

Reply:
It is best to avoid referring to data members where possible. Perhaps we can describe this method as "calls the StoreEntry abort handler (if any)" or something along those lines (I did not check the code when writing this -- just your summary)? If that is an accurate description, then it will expose the clash with the "client side" in the method name -- I can help with fixing that later. If that is not an accurate description, then I misinterpreted your comment summarizing what this method does.

Reply:
> since it is public

This method is public because it is called from a local static CallbackClientSide(). I think we can make CallbackClientSide() a static member and then callbackClientSide() will become private.

> Perhaps we can describe this method as "calls the StoreEntry abort handler..."

 store_client::Callback::callback_handler stores client-side STCB callbacks, such as clientReplyContext::HandleIMSReply(), clientReplyContext::CacheHit(), clientReplyContext::SendMoreData() and others. callbackClientSide() invokes them, passing the StoreIOBuffer buffer as a parameter. So we can document it as "invokes a previously stored STCB call" or something like that.

Reply:
> So we can document it as "invokes a previously stored STCB call" or something like that.

Aha, that clarifies things quite a bit, thank you! I suggest either "calls STCB" or "schedules an asynchronous STCB call", whichever is true. I will rename the method later and can polish its description further at that time (if needed).